pushjack/apns.py

```diff
@@ -416,7 +446,7 @@ def sendall(self, stream, error_timeout=10):
                 if ex.fatal:
                     # We can't continue due to a fatal error. Go ahead and
                     # convert remaining notifications to errors.
-                    errors += [APNSUnsendableError(i + ex.identifier)
+                    errors += [APNSUnsendableError(i + stream.next_identifier)
```

Comment:
I'm not sure if I got the logic you implemented here right, but from the needs of my push server implementation, it is necessary for me to know the reason for failure (if it's a timeout issue, or something else). In the latest commit of this PR, the problematic token exception is added to the errors list on line 446, and then if the exception is fatal, it will be left there untouched, and only the next 'unsendable' token exceptions will be added to the errors array. The result is we will still have all the errors in the list, but they won't all be 'unsendable'. Instead, it will preserve the fatal error which caused the failure.

Reply:
Again, your bug fix here is the right thing to do. Thanks!